sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1205,21 +1247,21 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                 ValueStopwatch cycleDuration;
                 var partitionIds = default(string[]);
 
-                while (!cancellationToken.IsCancellationRequested)
+                try
                 {
-                    cycleDuration = ValueStopwatch.StartNew();
+                    partitionIds = await connection.GetPartitionIdsAsync(RetryPolicy, cancellationToken).ConfigureAwait(false);
```

Comment:
Doesn't this need to stay inside the `while` block for `PerformLoadBalancingAsync` to work properly?

Reply:
I might be missing something but AFAIK it's impossible to change the partition count after the hub creation.

Reply:
We'll need to keep support for detecting changes to partitions.